src/txn/txn_log.c

```diff
@@ -268,18 +268,28 @@ __wt_txn_log_op(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt)
  *	Write the operations of a transaction to the log at commit time.
  */
 int
-__wt_txn_log_commit(WT_SESSION_IMPL *session, const char *cfg[])
+__wt_txn_log_commit(WT_SESSION_IMPL *session)
 {
 	WT_TXN *txn;
+	WT_TXN_GLOBAL *txn_global;
 
-	WT_UNUSED(cfg);
 	txn = &session->txn;
+	txn_global = &S2C(session)->txn_global;
+
 	/*
 	 * If there are no log records there is nothing to do.
 	 */
 	if (txn->logrec == NULL)
 		return (0);
 
+	/* Check if the commit is out of timestamp order. */
+	if (F_ISSET(txn, WT_TXN_PUBLIC_TS_COMMIT)) {
```

Comment:
We could additionally check `&& !txn_global->commit_ooo`.  Once it is set it doesn't get reset until the `log_flush` call and it may avoid repeated lock acquisitions, etc if there are a lot of ooo commits each time.